app/assets/javascripts/questionnaire_logic.js.erb

```diff
@@ -123,8 +143,9 @@ function disableHiddenInputs() {
 
 $(function () {
   timestamp_c = new Date().getTime();
-  $('input[type=radio],input[type=checkbox]').change(toggle_shown_questions).change(toggle_otherwise_field);
+  $('input[type=radio],input[type=checkbox]').change(toggle_shown_questions).change(toggle_otherwise_field).change(toggle_hidden_questions);
   $('input[type=radio][data-shows-questions],input[type=checkbox][data-shows-questions]').each(toggle_shown_questions);
+  $('input[type=radio][data-hides-questions],input[type=checkbox][data-hides-questions]').each(toggle_hidden_questions);
```

Comment:
je hoeft de DOM hier maar 1 keer te lezen (op line 147 kun je opslaan en nog een keer gebruiken). Dat lezen van die dom is best inefficient namelijk. Kun je die `.each` dingen ook chainen zoals je dat deed met de `.change`

Reply:
Nee want de selectors zijn verschillend. Die bovenste gaat over data-shows-questions dingen en die onderste over data-hides-questions